github/github.go

```diff
@@ -390,7 +390,7 @@ func parseRate(r *http.Response) Rate {
 // The provided ctx must be non-nil. If it is canceled or times out,
 // ctx.Err() will be returned.
 func (c *Client) Do(ctx context.Context, req *http.Request, v interface{}) (*Response, error) {
-	req = req.WithContext(ctx)
+	ctx, req = addContext(ctx, req)
```

Comment:
Yes, it has an effect on App Engine, and what App Engine does depends on what environment the code is run under.

Here is an example:
https://github.com/golang/appengine/blob/master/internal/api_classic.go#L56

Basically, it associates each App Engine request with a context so that logging (and other things) can be associated with a particular request.
When you are dealing with hundreds or thousands of queries per second, it is a really nice feature in the developers console to associate the log lines with the request that caused the them.
You can then mentally follow the flow of the single request.